code/game/turfs/simulated/minerals.dm

```diff
@@ -20,37 +20,51 @@
 	color = COLOR_ROCK
 	var/environment_type = "asteroid"
 	var/turf/simulated/floor/plating/turf_type = /turf/simulated/floor/plating/asteroid/airless
-	var/mineralType = null
-	var/mineralAmt = 3
-	var/spread = 0 //will the seam spread?
-	var/spreadChance = 0 //the percentile chance of an ore spreading to the neighboring tiles
 	var/last_act = 0
-	var/scan_state = "" //Holder for the image we display when we're pinged by a mining scanner
+
 	var/defer_change = 0
 	var/mine_time = 4 SECONDS //Changes how fast the turf is mined by pickaxes, multiplied by toolspeed
 	/// Should this be set to the normal rock colour on init?
 	var/should_reset_color = TRUE
 
+	/// The ore type, if any, that should spawn in the wall on Initialize.
+	/// Expected to be a subtype of [/datum/ore].
+	var/preset_ore_type
+	/// The representation of the unmined ore in the wall, if any.
+	var/datum/ore/ore
+
 /turf/simulated/mineral/Initialize(mapload)
 	. = ..()
+
 	if(should_reset_color)
 		color = COLOR_ROCK
 
-	if(mineralType && mineralAmt && spread && spreadChance)
-		for(var/dir in GLOB.cardinal)
-			if(prob(spreadChance))
-				var/turf/T = get_step(src, dir)
-				if(istype(T, /turf/simulated/mineral/random))
-					Spread(T)
+	if(preset_ore_type)
+		set_ore(preset_ore_type)
+
 	AddComponent(/datum/component/debris, DEBRIS_ROCK, -20, 10, 1)
 
-/turf/simulated/mineral/proc/Spread(turf/T)
-	T.ChangeTurf(type)
+/turf/simulated/mineral/proc/set_ore(ore_type)
+	if(!ore_type)
```

Comment:
Could you check if src already has an ore and delete the ore datum if it does?

Reply:
Done, PTAL